shopfloor_mobile_base/static/wms/src/components/datepicker/date_picker.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2022 Camptocamp SA (http://www.camptocamp.com)
+ * License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+ */
+
+import event_hub from "../../services/event_hub.js";
+
+export var DatePicker = Vue.component("date-picker-input", {
+    data: function () {
+        return {
+            date: "",
+        };
+    },
+    watch: {
+        date: function () {
+            this.$root.trigger("date_picker_selected", this.date);
+        },
+    },
+    mounted() {
+        event_hub.$on("datepicker:lotselected", (lot) => {
```

Comment:
:fire: 

This is a specific feature that should be implemented by the consumer of this widget.
You should have something like

`<date-picker-input v-on:date_selected="on_date_selected"...`

Reply:
Also, this change should be move to its own commit and separated